include/alpaka/mem/buf/BufOacc.hpp

```diff
@@ -0,0 +1,650 @@
+                return (*pitch).m_pitchBytes[TIdxIntegralConst::value];
+            }
+        };
+
+        //#############################################################################
+        //! The BufOacc 1D memory allocation trait specialization.
+        template<
+            typename TElem,
+            typename TIdx>
+        struct BufAlloc<
+            TElem,
+            DimInt<1u>,
+            TIdx,
+            DevOacc>
+        {
+            //-----------------------------------------------------------------------------
+            template<
+                typename TExtent>
+            ALPAKA_FN_HOST static auto allocBuf(
+                DevOacc const & dev,
+                TExtent const & extent)
+            -> BufOacc<TElem, DimInt<1u>, TIdx>
+            {
+                ALPAKA_DEBUG_MINIMAL_LOG_SCOPE;
+
+                auto const width(extent::getWidth(extent));
+                auto const widthBytes(width * static_cast<TIdx>(sizeof(TElem)));
+
+                dev.makeCurrent();
+                void * memPtr = acc_malloc(static_cast<std::size_t>(widthBytes));
+
+#if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
+                std::cout << __func__
+                    << " ew: " << width
+                    << " ewb: " << widthBytes
+                    << " ptr: " << memPtr
+                    << " device: " << dev.m_spDevOaccImpl->iDevice()
+                    << std::endl;
+#endif
+                return
+                    BufOacc<TElem, DimInt<1u>, TIdx>(
+                        dev,
+                        reinterpret_cast<TElem *>(memPtr),
+                        extent);
+            }
+        };
+
+        //#############################################################################
+        //! The BufOacc nD memory allocation trait specialization. \todo Add pitch
+        template<
+            typename TElem,
+            typename TDim,
+            typename TIdx>
+        struct BufAlloc<
+            TElem,
+            TDim,
+            TIdx,
+            DevOacc>
+        {
+            //-----------------------------------------------------------------------------
+            template<
+                typename TExtent>
+            ALPAKA_FN_HOST static auto allocBuf(
+                DevOacc const & dev,
+                TExtent const & extent)
+            -> BufOacc<TElem, TDim, TIdx>
+            {
+                ALPAKA_DEBUG_MINIMAL_LOG_SCOPE;
+
+                const std::size_t size = static_cast<std::size_t>(extent.prod()) * sizeof(TElem);
+
+                dev.makeCurrent();
+                void * memPtr = acc_malloc(size);
+#if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
+                std::cout << __func__ << "alloc'd " << TDim::value
+                    << "D device ptr: " << memPtr << " on device " << dev.m_spDevOaccImpl->iDevice()
+                    << " size " << size << " = " << static_cast<std::size_t>(extent::getExtentVec(extent).prod())*sizeof(TElem) << '\n';
```

Comment:
Why do you print the size twice?
```suggestion
                    << " size: " << size << '\n';
```

Reply:
Could you apply this to the OMP5 implementation as well please?

Reply:
Maybe to check if the correct extent vector was returned... . This is very early debug output.